scripts/generate_dump

```diff
@@ -1511,9 +1529,49 @@ while getopts ":xnvhzas:t:r:d" opt; do
             ;;
         /?)
             echo "Invalid option: -$OPTARG" >&2
-            exit 1
+            exit $EXT_GENERAL
             ;;
     esac
 done
 
-main
+# Check permissions before proceeding further
+if [ `whoami` != root ] && ! $NOOP;
+then
+    echo "$0: must be run as root (or in sudo)" >&2
+    exit 10
+fi
+
+##
+## Attempt Locking
```

Comment:
>Attempt Locking [](http://example.com/codeflow?start=3&length=15)

Did you try a simpler solution [https://unix.stackexchange.com/a/398710/161486](https://unix.stackexchange.com/a/398710/161486)?

Reply:
That should work for the most part (there can be false-positives though such as when an editor opens the text file). The main concern however is the pidof is not atomic but mkdir is

Reply:
What "atomic" matters? If some process is running, just exit.

Reply:
If let's say two tech-support processes reach `pidof` at the same time, both of them read the other PID as the current "running" PID, and both of them exit. 

In the case of mkdir, atleast one of them is guaranteed to return success and one of these will return failure and thus atleast one is guaranteed to run.